internal/schema/usage_data.go

```diff
@@ -149,8 +151,64 @@ func (u *UsageData) CalcEstimationSummary() map[string]bool {
 	return estimationMap
 }
 
-func NewUsageMap(m map[string]interface{}) map[string]*UsageData {
-	usageMap := make(map[string]*UsageData)
+// UsageMap is a map of address to UsageData built from a usage file.
+// UsageMap is a standalone type so that we can do more involved matching functionality.
+type UsageMap map[string]*UsageData
+
+// Get returns UsageData for a given resource address.
+func (usage UsageMap) Get(address string) *UsageData {
+	var data *UsageData
+
+	parsedAddress, err := addressParser.NewAddress(address)
+	if err == nil {
+		val, ok := usage[parsedAddress.ResourceSpec.Type]
+		if ok {
+			data = val
+		}
+	}
+
+	if ud := usage[address]; ud != nil {
+		if data != nil {
+			MergeAttributes(data, ud)
```

Comment:
It seems from the code that I extracted this from that we were only ever merging attributes from the `type` spec if this was a direct hit. I'm not sure if we should keep this behaviour, or merge the usage data if we also get a hit on the wildcard matches. I think we probably should, but I'm unsure why this wasn't done in the first place?

Reply:
Discussed on zoom, this is a bug.